sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/models/HttpHeaders.java

```diff
@@ -86,14 +97,49 @@ public HttpHeaders add(HttpHeaderName name, String value) {
         return addInternal(name.getCaseInsensitiveName(), name.getCaseSensitiveName(), value);
     }
 
+    /**
+     * Adds a {@link HttpHeader} with the given name and value if a {@link HttpHeader} with that name doesn't already exist,
+     * otherwise adds the {@code values} to the existing header.
+     *
+     * @param name The name of the {@link HttpHeader}.
+     * @param values The values of the {@link HttpHeader}.
+     * @return The updated {@link HttpHeaders} object.
+     */
+    public HttpHeaders add(HttpHeaderName name, List<String> values) {
+        return addInternal(name.getCaseInsensitiveName(), name.getCaseSensitiveName(), values);
+    }
+
+    /**
+     * Adds all provided {@link HttpHeader} key/values pairs into this {@link HttpHeaders} instance. This is equivalent to
+     * calling {@code headers.forEach(this::add)}. This will create a {@link HttpHeader} for each key with the given
+     * name and value if a {@link HttpHeader} with that name doesn't already exist,otherwise adds the {@code values}
+     * to the existing header. If the given values list is
+     * {@code null}, the header with the given name will be removed.
+     *
+     * <p>Use {@link #addAllHeaders(HttpHeaders)} if you already have an instance of {@link HttpHeaders} as it provides better
+     * performance.</p>
+     *
+     * @param headers A map containing keys representing {@link HttpHeader} names, and keys representing the associated
+     * values.
+     *
+     * @return The updated {@link HttpHeaders} object.
+     *
+     * @throws NullPointerException If {@code headers} is {@code null}.
+     */
+    public HttpHeaders addAll(Map<String, List<String>> headers) {
+        headers.forEach((name, value) -> addInternal(formatKey(name), name, value));
+
+        return this;
+    }
+
     private HttpHeaders addInternal(String formattedName, String name, String value) {
```

Comment:
Should this be changed to taking `HttpHeaderName` as well? Force APIs using `String` to convert the String to HttpHeaderName and the APIs already using HttpHeaderName just pass it as is.